galaxy_ng/app/api/ui/views/ai_index.py

```diff
@@ -0,0 +1,180 @@
+
+    @extend_schema(
+        request=inline_serializer(
+            name="AddToAIIndexDenyList",
+            fields={"reference": serializers.CharField()}
+        ),
+        parameters=[
+            OpenApiParameter(
+                "scope", OpenApiTypes.STR, "path", enum=["namespace", "legacy_namespace"]
+            )
+        ],
+        responses={
+            201: inline_serializer(
+                name="AddedToAIIndexDenyList",
+                fields={"scope": serializers.CharField, "reference": serializers.CharField},
+            ),
+            403: "Forbidden",
+            400: "Bad Request",
+            409: "Conflict"
+        },
+    )
+    @transaction.atomic
+    def post(self, request, scope, *args, **kwargs):
+        """Adds a collection to the AI Index deny list.
+
+        http::
+            POST _ui/v1/ai_index/{namespace|legacy_namespace}/
+            {
+                "reference": "some_name"
+            }
+
+        responses:
+            201: Created
+            400: Bad Request (invalid payload data)
+            409: Conflict (data already in the model)
+        """
+        data = {"scope": scope, "reference": request.data.get("reference")}
+        obj = AIIndexDenyList(**data)
+        try:
+            obj.full_clean(validate_unique=False)
+        except ValidationError as e:
+            return Response(e, status=status.HTTP_400_BAD_REQUEST)
+
+        self._verify_permission_for_namespace_object(request, **data)
+
+        try:
+            obj.save()
+        except IntegrityError as e:
+            return Response(str(e), status=status.HTTP_409_CONFLICT)
+
+        return Response(
+            {"scope": obj.scope, "reference": obj.reference},
+            status=status.HTTP_201_CREATED
+        )
+
+
+class AIIndexListView(AIIndexBaseView):
+    """Lists all entries in the AIIndexDenyList.
+    Open without auth.
+    """
+    action = "ai-index-list"
+
+    @extend_schema(
+        responses={
+            200: inline_serializer(
+                name="AIIndexDenyList",
+                fields={"items": serializers.ListField(child=serializers.CharField())}
+            )
+        },
+        parameters=[
+            OpenApiParameter("scope", OpenApiTypes.STR, enum=["namespace", "legacy_namespace"])
+        ]
+    )
+    def get(self, request, *args, **kwargs):
+        """Returns a list of all entries in the AI Index deny list.
+
+        http::
+            GET _ui/v1/ai_index/
+            GET _ui/v1/ai_index/?scope=namespace
+
+        responses:
+            200: Ok {items: ["..."]}
+
+        Query params:
+            scope: "namespace" or "legacy_namespace" to filter
+
+        Sorted by reference field.
+        """
+        qs = AIIndexDenyList.objects.all().order_by("reference")
+        if scope := request.GET.get("scope"):
+            qs = qs.filter(scope=scope)
+        items = qs.values_list("reference", flat=True)
+        return Response({"items": list(items)})
```

Comment:
This is returning

```json
{
    "items": [
        "foo",
        "bar", 
        "zaz"
    ]
}
```

And accepts a `?scope=` filter parameter, I think this is easier wherever it is consumed to parse, I am not sure if we want to serialize everything like:

```json
{
    "items": [
        {"scope": "namespace", "reference": "foo"},
        ...
    ]
}
```

@jctanner @awcrosby